src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -392,7 +392,7 @@ internal ImmutableArray<IVariableDeclaratorOperation> CreateVariableDeclarator(B
         private IPlaceholderOperation CreateBoundDeconstructValuePlaceholderOperation(BoundDeconstructValuePlaceholder boundDeconstructValuePlaceholder)
         {
             SyntaxNode syntax = boundDeconstructValuePlaceholder.Syntax;
-            ITypeSymbol type = boundDeconstructValuePlaceholder.Type;
+            ITypeSymbol type = boundDeconstructValuePlaceholder.Type.GetPublicSymbol();
```

Comment:
I think we can probably defer the realization of the symbols created for operations until requested from the IOperation, rather than realizing them up front.

Reply:
> I think we can probably defer the realization of the symbols created for operations until requested from the IOperation, rather than realizing them up front.

I will open a follow up issue for this because this is a big work item on its own and the end benefit is not obvious.

---
In reply to: [339210407](https://github.com/dotnet/roslyn/pull/39498#discussion_r339210407) [](ancestors = 339210407)